core/store/src/deserialized_column.rs

```diff
@@ -41,7 +44,19 @@ impl ColumnCache {
     fn new(capacity: usize) -> Option<Mutex<Self>> {
         let capacity = NonZeroUsize::new(capacity);
         let values = capacity.map(|cap| lru::LruCache::new(cap))?;
-        Some(Mutex::new(Self { values, active_flushes: 0 }))
+        Some(Mutex::new(Self { values, active_flushes: 0, store_none_values: false }))
+    }
+
+    fn with_none_values(inner: Option<Mutex<Self>>) -> Option<Mutex<Self>> {
+        if let Some(cache) = &inner {
+            let mut cache = cache.lock();
+            cache.store_none_values = true;
+        };
+        inner
+    }
+
+    pub(super) fn store_none_values(&self) -> bool {
```

Comment:
made this to make sure the field is read-only, we can remove it and directly reference the field if desired.